js/private/js_info.bzl

```diff
@@ -31,10 +32,33 @@ def js_info(
         transitive_types: See JsInfo documentation
         npm_sources: See JsInfo documentation
         npm_package_store_infos: See JsInfo documentation
+        **kwargs: For backward compat support
 
     Returns:
         A JsInfo provider
     """
+
+    # Handle backward compat for rules_js 1.x js_info factory parameters:
+    # - declarations
+    # - transitive_declarations
+    # - npm_package_store_deps
+    # - transitive_npm_linked_package_files
+    declarations = kwargs.pop("declarations", None)
+    if declarations != None:
+        types = declarations
+    transitive_declarations = kwargs.pop("transitive_declarations", None)
+    if transitive_declarations != None:
+        transitive_types = transitive_declarations
+    npm_package_store_deps = kwargs.pop("npm_package_store_deps", None)
+    if npm_package_store_deps != None:
+        npm_package_store_infos = npm_package_store_deps
+    transitive_npm_linked_package_files = kwargs.pop("transitive_npm_linked_package_files", None)
+    if transitive_npm_linked_package_files != None:
+        npm_sources = transitive_npm_linked_package_files
+    if len(kwargs):
+        msg = "Invalid js_info parameter '{}'".format(kwargs.keys()[0])
+        fail(msg)
+
```

Comment:
Should we record which legacy names we `pop`ed and output a warning here?

Reply:
Updated. Warning per attribute will work.